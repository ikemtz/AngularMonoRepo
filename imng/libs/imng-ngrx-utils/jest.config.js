module.exports = {
  name: 'imng-ngrx-utils',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/libs/imng-ngrx-utils',
  snapshotSerializers: [
    'jest-preset-angular/AngularSnapshotSerializer.js',
    'jest-preset-angular/HTMLCommentSerializer.js'
  ]
};
