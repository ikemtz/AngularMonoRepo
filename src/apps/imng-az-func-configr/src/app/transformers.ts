import { IAzSetting, ILocalSetting } from './az-setting.model';

export interface ITransformer {
    icon: string;
    name: string;
    id: number;
    convert(data: IAzSetting[]): ILocalSetting | string;
}
export const initialFunctionDevSettings = (): ILocalSetting => ({ IsEncrypted: false, Values: {} });
export const AzFunc: ITransformer = {
    // eslint-disable-next-line max-len
    icon: '//raw.githubusercontent.com/Azure/azure-functions-cli/master/src/Azure.Functions.Cli/npm/assets/azure-functions-logo-color-raster.png',
    name: 'AZ Function',
    id: 1,
    convert: data => {
        const devSettings = initialFunctionDevSettings();
        if (Array.isArray(data)) {
            data
                .map(t => ({ name: t.name, value: t.value }))
                .forEach(t => {
                    devSettings.Values[t.name] = t.value;
                });
        }
        return devSettings;
    }
};

export const AzWebApp: ITransformer = {

    icon: '//4.bp.blogspot.com/-BmgLupDPg-s/V-cGGu3UiwI/AAAAAAAAL70/m4SD_XfwO-Y4Z4n_55ZqrsXfFLa2slzTwCEw/s1600/Azure%2BApp%2BService%2B-%2BWeb%2BApp%2B%2528was%2BWebsites%2529.png',
    name: 'AZ WebApp',
    id: 2,
    convert: data => {
        const devSettings = {};
        if (Array.isArray(data)) {
            data
                .map(t => ({ name: t.name, value: t.value }))
                .forEach(t => {
                    devSettings[t.name] = t.value;
                });
        }
        return devSettings;
    }
};

export const DockerEnv: ITransformer = {
    icon: '//www.docker.com/sites/default/files/d8/2019-07/Moby-logo.png',
    name: 'Docker Environment',
    id: 3,
    convert: data => Array.isArray(data) ? data.map(t => `${t.name}=${t.value}`).join('\n\r') : ''
};

export const transformers: ITransformer[] = [AzFunc, AzWebApp, DockerEnv];
