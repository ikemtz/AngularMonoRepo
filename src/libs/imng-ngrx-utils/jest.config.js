module.exports = {
  name: 'imng-ngrx-utils',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/libs/imng-ngrx-utils',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
