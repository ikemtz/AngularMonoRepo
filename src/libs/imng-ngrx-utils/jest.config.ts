import type { Config } from 'jest';

const config: Config = {
  displayName: 'imng-ngrx-utils',
  reporters: [
    'default',
    [
      'jest-junit',
      {
        outputDirectory: '.',
        outputName: './junit.imng-ngrx-utils.xml',
      },
    ],
  ],
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {},
  coverageDirectory: '../../coverage/libs/imng-ngrx-utils',
  transform: {
    '^.+\\.(ts|mjs|js|html)$': [
      'jest-preset-angular',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
        stringifyContentPathRegex: '\\.(html|svg)$',
      },
    ],
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
  preset: '../../jest.preset.js',
};

export default config;
