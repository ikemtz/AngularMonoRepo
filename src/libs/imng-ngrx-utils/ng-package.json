{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/imng-ngrx-utils",
  "lib": {
    "entryFile": "src/lib/public_api.ts"
  },
  "deleteDestPath": true
}
