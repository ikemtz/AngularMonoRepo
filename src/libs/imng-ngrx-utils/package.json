{
  "name": "imng-ngrx-utils",
  "version": "0.0.4",
  "author": "Isaac Martinez <@ikemtz>",
  "license": "MIT",
  "peerDependencies": {
    "@angular/common": "^8.0.0",
    "@angular/core": "^8.0.0",
    "@progress/kendo-data-query": "^1.5.1",
    "@ngrx/store": "^8.0.1"
  },
  "devDependencies": {},
  "ngPackage": {
    "$schema": "./node_modules/ng-packagr/ng-package.schema.json",
    "lib": {
      "entryFile": "src/index.ts"
    },
    "dest": "../../dist/libs/imng-ngrx-utils"
  }
}
