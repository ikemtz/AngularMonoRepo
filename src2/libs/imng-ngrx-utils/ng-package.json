{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/imng-ngrx-utils",
  "lib": {
    "entryFile": "src/index.ts"
  }
}
