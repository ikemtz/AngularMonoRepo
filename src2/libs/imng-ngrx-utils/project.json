{
  "projectType": "library",
  "root": "libs/imng-ngrx-utils",
  "sourceRoot": "libs/imng-ngrx-utils/src",
  "prefix": "imng",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:package",
      "outputs": ["dist/libs/imng-ngrx-utils"],
      "dependsOn": [{ "target": "build", "projects": "dependencies" }],
      "options": {
        "project": "libs/imng-ngrx-utils/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/imng-ngrx-utils"],
      "dependsOn": [{ "target": "build", "projects": "dependencies" }],
      "options": {
        "jestConfig": "libs/imng-ngrx-utils/jest.config.js",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/imng-ngrx-utils/src/**/*.ts",
          "libs/imng-ngrx-utils/src/**/*.html",
          "libs/imng-ngrx-utils/testing/**/*.ts",
          "libs/imng-ngrx-utils/testing/**/*.html"
        ]
      }
    }
  },
  "tags": []
}
