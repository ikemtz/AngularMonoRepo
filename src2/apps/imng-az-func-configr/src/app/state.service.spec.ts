/* eslint-disable @typescript-eslint/no-explicit-any */
import { TestBed } from '@angular/core/testing';
import { StateService } from './state.service';
import { AzFunc, AzWebApp, DockerEnv } from './transformers';
import { readFirst } from 'imng-ngrx-utils/testing';
import { IAzSetting } from './az-setting.model';

describe('StateService', () => {
  let service: StateService;
  const azConfigFactory = (): IAzSetting[] => [
    { name: 'dbConnection', value: 'MySqlServer', slotSetting: false }
  ];

  beforeEach(() => {
    TestBed.configureTestingModule({});
    service = TestBed.inject(StateService);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  it('should setTransformer AzFunc', async () => {
    const transformer = AzFunc;
    service.setTransformer(transformer);
    service.setAzConfiguration(azConfigFactory());
    const result = await readFirst(service.currentTransformer$);
    const data = await readFirst(service.output$);
    expect(result).toEqual(transformer);
    expect(data).toMatchSnapshot();
  });

  it('should setTransformer AzWebApp', async () => {
    const transformer = AzWebApp;
    service.setTransformer(transformer);
    service.setAzConfiguration(azConfigFactory());
    const result = await readFirst(service.currentTransformer$);
    const data = await readFirst(service.output$);
    expect(result).toEqual(transformer);
    expect(data).toMatchSnapshot();
  });

  it('should setTransformer DockerEnv', async () => {
    const transformer = DockerEnv;
    service.setTransformer(transformer);
    service.setAzConfiguration(azConfigFactory());
    const result = await readFirst(service.currentTransformer$);
    const data = await readFirst(service.output$);
    expect(result).toEqual(transformer);
    expect(data).toMatchSnapshot();
  });


  it('should setTransformer AzFunc Empty', async () => {
    const transformer = AzFunc;
    service.setAzConfiguration('null' as any);
    service.setAzConfiguration('null' as any);
    service.setAzConfiguration('null' as any);
    const result = await readFirst(service.currentTransformer$);
    const data = await readFirst(service.output$);
    expect(result).toEqual(transformer);
    expect(data).toMatchSnapshot();
  });

  it('should setTransformer AzWebApp Empy', async () => {
    const transformer = AzWebApp;
    service.setTransformer(transformer);
    service.setAzConfiguration('' as any);
    const result = await readFirst(service.currentTransformer$);
    const data = await readFirst(service.output$);
    expect(result).toEqual(transformer);
    expect(data).toMatchSnapshot();
  });

  it('should setTransformer DockerEnv Empty', async () => {
    const transformer = DockerEnv;
    service.setTransformer(transformer);
    service.setAzConfiguration('' as any);
    const result = await readFirst(service.currentTransformer$);
    const data = await readFirst(service.output$);
    expect(result).toEqual(transformer);
    expect(data).toMatchSnapshot();
  });
});
