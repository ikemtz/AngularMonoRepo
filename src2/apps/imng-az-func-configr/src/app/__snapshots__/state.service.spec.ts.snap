// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateService should setTransformer AzFunc 1`] = `
Object {
  "IsEncrypted": false,
  "Values": Object {
    "dbConnection": "MySqlServer",
  },
}
`;

exports[`StateService should setTransformer AzFunc Empty 1`] = `
Object {
  "IsEncrypted": false,
  "Values": Object {},
}
`;

exports[`StateService should setTransformer AzWebApp 1`] = `
Object {
  "dbConnection": "MySqlServer",
}
`;

exports[`StateService should setTransformer AzWebApp Empy 1`] = `Object {}`;

exports[`StateService should setTransformer DockerEnv 1`] = `"dbConnection=MySqlServer"`;

exports[`StateService should setTransformer DockerEnv Empty 1`] = `""`;
